client/app/page.tsx

```diff
@@ -1,26 +1,68 @@
-import React from "react";
+'use client'
+
+import axios from 'axios'
+import { useSearchParams } from 'next/navigation'
+import { Suspense, useEffect, useState } from 'react'
+
+export default function HomePage() {
+  return (
+    <Suspense fallback={<div>Loading...</div>}>
+      <HomeContent />
+    </Suspense>
+  )
+}
+
+function HomeContent() {
+  const [token, setToken] = useState('')
+
+  const queryParams = useSearchParams()
+  const code = queryParams.get('code')
+  const state = queryParams.get('state')
+  const client_redirect_uri = 'http://localhost:80'
```

Comment:
Should this port be 8080? This variable could also be out of this function

Reply:
This is the client URI, not server one, so 80 is fine

Reply:
Realized there's two questions so unresolving for the second part. 80 is still correct though

Reply:
Ok thanks